main.tf

```diff
@@ -80,25 +80,31 @@ data "aws_iam_policy_document" "service_account_assume_role" {
 
     principals {
       type        = "Federated"
-      identifiers = [format("arn:%s:iam::%s:oidc-provider/%s", var.aws_partition, local.aws_account_number, local.eks_cluster_oidc_issuer)]
+      identifiers = [for url in local.eks_cluster_oidc_issuers : format("arn:%s:iam::%s:oidc-provider/%s", var.aws_partition, local.aws_account_number, url)]
     }
 
-    condition {
-      test     = "StringLike"
-      values   = formatlist("system:serviceaccount:%s", local.service_account_namespace_name_list)
-      variable = format("%s:sub", local.eks_cluster_oidc_issuer)
-
+    dynamic "condition" {
+      for_each = local.eks_cluster_oidc_issuers
+      content {
+        test     = "StringLike"
+        values   = formatlist("system:serviceaccount:%s", local.service_account_namespace_name_list)
+        variable = format("%s:sub", condition.value)
+      }
     }
-    condition {
-      test     = "StringEquals"
-      values   = ["sts.amazonaws.com"]
-      variable = format("%s:aud", local.eks_cluster_oidc_issuer)
+
+    dynamic "condition" {
+      for_each = local.eks_cluster_oidc_issuers
+      content {
+        test     = "StringEquals"
+        values   = ["sts.amazonaws.com"]
+        variable = format("%s:aud", condition.value)
+      }
     }
```

Comment:
I'm leaning against adding this feature altogether, but if we do add it, I'm pretty sure this is wrong. By adding a separate condition for each OIDC issuer, you are requiring all the ODICs to validate the role, when only one will. See [Condition Logic](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-logic-multiple-context-keys-or-values.html#reference_policies_multiple-conditions-eval) for details. 

I believe the correct solution is to loop over `statement` rather than conditions, and have on statement for each OIDC provider.